bookwyrm/static/js/bookwyrm.js

```diff
@@ -368,4 +374,84 @@ let BookWyrm = new class {
             );
         }
     }
+
+    handleModalButton(element) {
+        const modalButton = element.currentTarget;
+        const targetModalId = modalButton.dataset.modalOpen;
+        const htmlElement = document.querySelector('html');
+        const modal = document.getElementById(targetModalId);
+        
+        if (!modal) {
+            return;
+        }
+
+        // Helper functions
+        function handleModalOpen(modalElement) {
+            htmlElement.classList.add('is-clipped');
+            modalElement.classList.add('is-active');
+            modalElement.getElementsByClassName('modal-card')[0].focus();
+
+            const closeButtons = modalElement.querySelectorAll("[data-modal-close]");
+            
+            closeButtons.forEach((button) => {
+                button.addEventListener(
+                    'click',
+                    function() { handleModalClose(modalElement) }
+                );
+            });
+
+            document.addEventListener(
+                'keydown',
+                function(event) {
+                    if (event.key === 'Escape') {
+                        handleModalClose(modalElement);
+                    }
+                }
+            );
+
+            modalElement.addEventListener('keydown', handleFocusTrap)
+        }
+
+        function handleModalClose(modalElement) {
+            modalElement.removeEventListener('keydown', handleFocusTrap)
+            htmlElement.classList.remove('is-clipped');
+            modalElement.classList.remove('is-active');
+            modalButton.focus();
+        }
+
+        function handleFocusTrap(event) {
+            const focusableEls = event.currentTarget.querySelectorAll(
+                [
+                    'a[href]:not([disabled])',
+                    'button:not([disabled])',
+                    'textarea:not([disabled])',
+                    'input[type="text"]:not([disabled])',
+                    'input[type="radio"]:not([disabled])',
+                    'input[type="checkbox"]:not([disabled])',
```

Comment:
I don't know querySelector super well, but it seems like is going to miss inputs of types `date`, `password`, and `number`. I wonder if it's possible to select all inputs that are _not_ of type `hidden`?